### PR TITLE
[Run2_UL] Change from absolute to signed impact parameters

### PR DESCRIPTION
**Pull request by aperloff**
- Switch from using built-in impact parameters w.r.t. the detector center, (0,0,0), to using the ones w.r.t. the primary vertex position.
- Switch from absolute to signed impact parameters
- Add back in some EMJ unit tests

These changes should have no effect on the size of the ntuples and no effect on the speed of the code.

### CODE REVIEW

File: Production/test/unitTest.py [hunk 1]
```diff
@@ -99,6 +99,10 @@ def defineTests(mytests, scenario, name, numevents, command, dataset, inputFiles
     mytests.append(Test("Summer20UL18","Summer20UL18.TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8",numevents,command,inputFilesConfig="Summer20UL18.TTToSemiLeptonic_TuneCP5_13TeV-powheg-pythia8",nstart=0,nfiles=1))
     mytests.append(Test("Summer20UL16","Summer20UL16.GJets_HT-200To400_TuneCP5_13TeV-madgraphMLM-pythia8",numevents,command,inputFilesConfig="Summer20UL16.GJets_HT-200To400_TuneCP5_13TeV-madgraphMLM-pythia8",nstart=0,nfiles=1))
     mytests.append(Test("Summer20UL16APV","Summer20UL16APV.GJets_HT-200To400_TuneCP5_13TeV-madgraphMLM-pythia8",numevents,command,inputFilesConfig="Summer20UL16APV.GJets_HT-200To400_TuneCP5_13TeV-madgraphMLM-pythia8",nstart=0,nfiles=1))
+    mytests.append(Test("Summer20UL16sig","PrivateSamples.EMJ_2016_mMed-1000_mDark-20_kappa-0p25_aligned-down",numevents,"emerging=True doPhotons=True deepAK8=False deepDoubleB=False doZinv=False nestedVectors=False splitLevel=99"+command,inputFilesConfig="PrivateSamples.EMJ_2016_mMed-1000_mDark-20_kappa-0p25_aligned-down",nstart=0,nfiles=10))
```
Comment (kpedro88) [inline Production/test/unitTest.py]: how does this list of samples interact with #595?

File: Utils/src/CandidateTrackFilter.cc [hunk 2]
```diff
@@ -119,13 +116,21 @@ class TrackInfos {
 		trks_quality->push_back(track.qualityMask());
 		// impact parameter variables
 		if( primaryVertex ){
-			std::pair<bool,Measurement1D> ip2d = IPTools::absoluteTransverseImpactParameter(transientTrack, *primaryVertex);
-			std::pair<bool,Measurement1D> ip3d = IPTools::absoluteImpactParameter3D(transientTrack, *primaryVertex);
+			auto primaryVertexPos = primaryVertex->position();
```
Comment (kpedro88) [inline Utils/src/CandidateTrackFilter.cc]: `const auto&`

File: Utils/src/CandidateTrackFilter.cc [hunk 3]
```diff
@@ -119,13 +116,21 @@ class TrackInfos {
...
+			trks_dxypv->push_back(track.dxy(primaryVertexPos));
+			trks_dxyerrorpv->push_back(track.dxyError(primaryVertexPos, primaryVertex->error()));
+			trks_dzpv->push_back(track.dz(primaryVertexPos)); //returns the ip wrt PV[0]
+			Global3DVector dir(track.px(), track.py(), track.pz());
+			std::pair<bool,Measurement1D> ip2d = IPTools::signedTransverseImpactParameter(transientTrack, dir, *primaryVertex);
```
Comment (kpedro88) [inline Utils/src/CandidateTrackFilter.cc]: `const auto&`

File: Utils/src/CandidateTrackFilter.cc [hunk 4]
```diff
@@ -119,13 +116,21 @@ class TrackInfos {
...
+			std::pair<bool,Measurement1D> ip3d = IPTools::signedImpactParameter3D(transientTrack, dir, *primaryVertex);
```
Comment (kpedro88) [inline Utils/src/CandidateTrackFilter.cc]: `const auto&`

File: Production/test/unitTest.py (hunk 1 above)
Comment (aperloff) [reply Production/test/unitTest.py]: These are the old samples, for now. These should be updated in #595 or updated after #595 is merged.

**Comment by kpedro88** [comment]
needs rebase after #595 merged

**Pull request merged**